src/System.Text.Json/tests/Serialization/DictionaryTests.KeyPolicy.cs

```diff
@@ -99,5 +101,38 @@ public static void NullNamePolicy()
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<Dictionary<string, int>>(@"{""onlyKey"": 1}", options));
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(new Dictionary<string, int> { { "onlyKey", 1 } }, options));
         }
+
+        [Fact]
+        public static void KeyConflictDeserialize_LastValueWins()
```

Comment:
@JamesNK  could you have a look to see if these scenarios look good? @layomia realized we're missing a feature equivalent for https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_Serialization_NamingStrategy_ProcessExtensionDataNames.htm, but that could come in the future